src/core/imports/mpi.cpp

```diff
@@ -1014,135 +1102,116 @@ void TaggedSendRecv
     Deserialize( rc, packedRecv, rbuf );
 }
 
-template<typename T>
-void SendRecv
-( const T* sbuf, int sc, int to,
-        T* rbuf, int rc, int from, Comm comm )
+template <typename T, Device D>
+void SendRecv(
+    const T* sbuf, int sc, int to,
+    T* rbuf, int rc, int from, Comm comm,
+    SyncInfo<D> const& syncInfo)
 EL_NO_RELEASE_EXCEPT
-{ TaggedSendRecv( sbuf, sc, to, 0, rbuf, rc, from, ANY_TAG, comm ); }
+{ TaggedSendRecv( sbuf, sc, to, 0, rbuf, rc, from, ANY_TAG, comm, syncInfo ); }
 
-template<typename T>
-T TaggedSendRecv( T sb, int to, int stag, int from, int rtag, Comm comm )
+template <typename T, Device D>
+T TaggedSendRecv( T sb, int to, int stag, int from, int rtag, Comm comm,
+                  SyncInfo<D> const& syncInfo)
 EL_NO_RELEASE_EXCEPT
 {
     T rb;
-    TaggedSendRecv( &sb, 1, to, stag, &rb, 1, from, rtag, comm );
+    TaggedSendRecv( &sb, 1, to, stag, &rb, 1, from, rtag, comm, syncInfo );
     return rb;
 }
 
-template<typename T>
-T SendRecv( T sb, int to, int from, Comm comm )
+template <typename T, Device D>
+T SendRecv( T sb, int to, int from, Comm comm, SyncInfo<D> const& syncInfo )
 EL_NO_RELEASE_EXCEPT
-{ return TaggedSendRecv( sb, to, 0, from, ANY_TAG, comm ); }
+{ return TaggedSendRecv( sb, to, 0, from, ANY_TAG, comm, syncInfo ); }
 
-template<typename Real,
+template <typename Real, Device D,
          typename/*=EnableIf<IsPacked<Real>>*/>
-void TaggedSendRecv
-( Real* buf, int count, int to, int stag, int from, int rtag, Comm comm )
+void TaggedSendRecv(
+    Real* buf, int count, int to, int stag, int from, int rtag, Comm comm,
+    SyncInfo<D> const& syncInfo)
 EL_NO_RELEASE_EXCEPT
 {
     EL_DEBUG_CSE
+
+    LogicError("SendRecv_replace. Tom want's to know if this is used.");
```

Comment:
I run into this error when I'm copying between non-aligned matrices.

Reply:
Sweet! I'll remove the error. I was just curious.